display_list/display_list_image_filter.h

```diff
@@ -534,6 +570,14 @@ class DlComposeImageFilter final : public DlImageFilter {
                                    inner_->skia_object());
   }
 
+  MatrixCapability matrix_capability() const override {
+    return MatrixCapability::kComplex;
```

Comment:
Another reason the vector of filters is overkill. Compose needs to return the least restrictive value here so that when its outer and inner are consulted and min-ified this value doesn't interfere with their answers.

Compare that to just having this method do "min(outer..., inner...)" and getting rid of the filters method.